orttraining/orttraining/training_ops/cuda/tensor/gather_grad.cc

```diff
@@ -16,7 +16,8 @@ ONNX_OPERATOR_KERNEL_EX(
     KernelDefBuilder()
         .InputMemoryType<OrtMemTypeCPUInput>(0)
         .TypeConstraint("I", DataTypeImpl::GetTensorType<int64_t>())
-        .TypeConstraint("T", DataTypeImpl::AllFixedSizeTensorTypes())
+        .TypeConstraint("T", {DataTypeImpl::GetTensorType<float>(),
+                              DataTypeImpl::GetTensorType<MLFloat16>()})
```

Comment:
CUDA should support float16

Reply:
> CUDA should support float16

`MLFloat16` is float16, no?
The current implementation only works for `float` and `MLFloat16`, that's why I changed the registration to these types:
https://github.com/microsoft/onnxruntime/blob/45dae88a1ad7c4878f4ba41ad54e5b0c069076e8/orttraining/orttraining/training_ops/cuda/tensor/gather_grad.cc#L77